Import nouveau (#4291)
apache/couchdb: java/nouveau/server/README.md

# nouveau
Lucene 9 + DropWizard = Maybe a good search option for Apache CouchDB?

Nouveau is an experimental search extension for CouchDB 3.x.

## What works?

* you can define a default analyzer and different analyzers by field name.
* sorting on text and numbers
* classic lucene query syntax
* count and range facets
* cursor support for paginating efficiently through large results sets
* indexes automatically deleted if database is deleted (as long as nouveau is running!)
* integration with ken
* update=false
* support for stale=ok
* integration with mango

## What doesn't work yet?

* include_docs=true
* No support for results grouping
* No support to configure stop words for analyzers

## Why is this better than dreyfus/clouseau?

* No scalang (or Scala!)
* Supports any version of Java that Lucene 9 supports
* memory-mapped I/O for performance
* direct I/O used for segment merging (so we don't evict useful data from disk cache)
* It's new and shiny.

> ## Erlang side
> 
> You'll need to run a fork of couchdb: https://github.com/rnewson/couchdb-nouveau
> 
> ## Getting started
> 
> Build Nouveau with;
> 
> `mvn package`
> 
> Run Nouvea with;
> 
> `java -jar target/nouveau-*.jar server nouveau.yaml`
> 
> Now run CouchDB using the 'nouveau' branch of my fork at https://github.com/rnewson/couchdb-nouveau;

Reviewer: ```suggestion
## Getting started

Build and run Nouveau with;

`make nouveau-start`

Now run CouchDB;
```
